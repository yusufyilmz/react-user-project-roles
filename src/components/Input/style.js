
import styled from 'styled-components';

export const Label = styled.label`
    float: left;
    width: 25%;
    margin: 15px 5px;
    font-size: x-large;
    `

Label.displayName = 'Label'


export const Input = styled.input`
    margin-left: auto;
    margin-right: auto;
    width: 100%;
    border: 1px solid #ccc;
    border-radius: 4px;
    resize: vertical;
    background-color: ${props => (props.type === 'submit' || props.type === 'button') && '#27435c'};
    display: ${props => (props.type === 'submit' || props.type === 'button') && 'flex'};
    justify-content: ${props => (props.type === 'submit' || props.type === 'button') && 'center'};
    color: ${props => (props.type === 'submit' || props.type === 'button') && 'white'};
    height: ${props => (props.type === 'submit' || props.type === 'button') && '50px'};
    text-indent: ${props => (props.type === 'text' ) && '30px'};
    font-size: ${props => (props.type === 'submit' || props.type === 'button') ? 'x-large' : 'medium' };
    padding: ${props => (props.type === 'submit' || props.type === 'button') ? '' : '15px 0' };

    `;

Input.displayName = 'Input'