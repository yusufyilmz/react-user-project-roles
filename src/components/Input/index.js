import React, { Fragment } from 'react';
import { Input, Label } from './style';


export default ({ type, name, text, onChange, value, onClick, required, placeHolder, width}) => {
    return (
        <Fragment>
            <Label htmlFor={name} >{text}</Label>
            <Input
                placeholder={placeHolder}
                width={width}
                name={name}
                autoComplete="off"
                required={required}
                id={name}
                type={type}
                onChange={onChange}
                onClick={onClick}
                value={value}>
            </Input>
        </Fragment>
    )
}


