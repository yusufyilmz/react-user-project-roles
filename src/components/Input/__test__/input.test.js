

import React from 'react';
import Input from '../index';
import { shallow } from 'enzyme'


describe('Input component', () => {

    let props, wrapper;

    beforeEach(() => {
        props = {
            text : 'test',
            onChange : jest.fn(),
            type : 'text',
            value : 'value' 
        }
     
        wrapper = shallow(
            <Input {...props}/>
        );
    });

    test('renders onclick correctly', () => {
        expect(wrapper.find('Label').text()).toBe(props.text);
    });

    test('renders onclick correctly', () => {
        expect(wrapper.find('Input').props()['value']).toBe(props.value);
    });

    test('renders onclick correctly', () => {
        wrapper.find('Input').simulate('change');
        expect(props.onChange.mock.calls.length).toEqual(1);
    });

    test('snapchat correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
})
