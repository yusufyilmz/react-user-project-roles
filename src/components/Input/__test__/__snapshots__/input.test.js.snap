// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component snapchat correctly 1`] = `
<Fragment>
  <Label>
    test
  </Label>
  <Input
    autoComplete="off"
    onChange={[MockFunction]}
    type="text"
    value="value"
  />
</Fragment>
`;
